### PR TITLE
MOSIP-23370 MOSIP -23371 Heathcheck consolidator and websub

### PR DESCRIPTION
**Pull request by urviljoshi**

### CODE REVIEW

File: consolidator/modules/kafkahealthcheck/kafkahealthcheck.bal [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import ballerina/jballerina.java;
+import consolidatorService.config;
+import ballerina/log;
+
+public function getAllTopicsKafka() returns handle|error? {
+    handle bootStrapServer = java:fromString(config:KAFKA_BOOTSTRAP_NODE);
+    handle newMosipKafkaAdminClientResult = newMosipKafkaAdminClient(bootStrapServer);
+    log:printInfo("Checking if metadata Topics are present");
+    handle|error? result = trap getAllTopics(newMosipKafkaAdminClientResult);
```
Comment (vishwa-vyom) [inline consolidator/modules/kafkahealthcheck/kafkahealthcheck.bal:9]: In our case we will have many topics, is there any other lite way of checking ?

File: hub/hub_service.bal [hunk 2]
```diff
@@ -42,21 +40,29 @@ http:Service healthCheckService = service object {
         }
         healthcheck:HealthCheckResp diskSpace = {status: diskSpaceStatus, details: {diskSpaceMetaData}};
 
-        //kafka
-        string kafkaStatus = "DOWN";
-        kafka:TopicPartition[]|kafka:Error producerResult = conn:statePersistProducer->getTopicPartitions(config:REGISTERED_WEBSUB_TOPICS_TOPIC);
```
Comment (vishwa-vyom) [inline hub/hub_service.bal:47]: this looks lite, can we use this in consolidator ?

File: hub/modules/config/configurations.bal [hunk 3]
```diff
@@ -47,7 +47,7 @@ public configurable decimal POLLING_INTERVAL = 10;
 public configurable decimal GRACEFUL_CLOSE_PERIOD = 5;
 
 # The port that is used to start the hub
-public configurable int HUB_PORT = 9000;
+public configurable int HUB_PORT = 9191;
```
Comment (vishwa-vyom) [inline hub/modules/config/configurations.bal:50]: Why are we changing the hub port ?

File: hub/modules/config/configurations.bal [hunk 4]
```diff
@@ -94,3 +94,12 @@ public configurable int KAFKA_CONSUMER_FETCH_MAX_BYTES = 3145728;
 
 # The maximum retry count
 public configurable int KAFKA_CONSUMER_MAX_PARTITION_FETCH_BYTES = 524288;
+
+# Kafka topic which is stores consolidated websub subscribers for this server
+public configurable string META_TOPICS = "registered-websub-topics,consolidated-websub-topics,registered-websub-subscribers,consolidated-websub-subscribers";
+
+# consolidator base url
+public configurable string CONSOLIDATOR_BASE_URL = "http://localhost:9192";
```
Comment (vishwa-vyom) [inline hub/modules/config/configurations.bal:102]: Can we keep the consolidator pod name here ?

File: hub/modules/inittopic/inittopic.bal [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import ballerina/jballerina.java;
```
Comment (vishwa-vyom) [inline hub/modules/inittopic/inittopic.bal:1]: Can we rename this to something else inittopic is not sounding good

Comment (vishwa-vyom) [reply hub/modules/inittopic/inittopic.bal:1]: we can call this internal_topic_helper, if this is ok lets change the method names, error message also to match

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: consolidator/modules/config/configurations.bal [hunk 6]
```diff
@@ -44,3 +44,14 @@ public configurable decimal POLLING_INTERVAL = 10;
 public configurable decimal GRACEFUL_CLOSE_PERIOD = 5;
 
 public final string CONSTRUCTED_CONSUMER_ID = util:generateRandomString();
+
+public final string CURRENT_WORKING_DIR = "user.dir";
+
+# The token validation URL of IDP
```
[the remark this answers is not in the document]
Comment (urviljoshi) [reply consolidator/modules/config/configurations.bal:50]: resolved

File: consolidator/consolidator_service.bal [hunk 7]
```diff
@@ -22,6 +22,68 @@ import consolidatorService.config;
...
+import consolidatorService.kafkahealthcheck;
+import ballerina/jballerina.java;
+
+http:Service healthCheckService = service object {
+
+    resource function get .() returns healthcheck:HealthCheckResp {
+        //diskspace
+        string diskSpaceStatus = "DOWN";
+        handle handleStr = java:fromString(config:CURRENT_WORKING_DIR);
+        handle fileObj = newFile(java:fromString(getCurrent(handleStr).toString()));
```
[the remark this answers is not in the document]
Comment (urviljoshi) [reply consolidator/consolidator_service.bal:36]: we are using the same code spring actuator uses

File: consolidator/modules/kafkahealthcheck/kafkahealthcheck.bal (hunk 1 above)
Comment (urviljoshi) [reply consolidator/modules/kafkahealthcheck/kafkahealthcheck.bal:9]: done

File: hub/hub_service.bal (hunk 2 above)
Comment (urviljoshi) [reply hub/hub_service.bal:47]: done in java

File: hub/modules/config/configurations.bal (hunk 3 above)
Comment (urviljoshi) [reply hub/modules/config/configurations.bal:50]: we use this as default there in our all env

File: hub/modules/inittopic/inittopic.bal (hunk 5 above)
Comment (urviljoshi) [reply hub/modules/inittopic/inittopic.bal:1]: done

File: hub/modules/config/configurations.bal (hunk 4 above)
Comment (urviljoshi) [reply hub/modules/config/configurations.bal]: done

File: consolidator/modules/config/configurations.bal (hunk 6 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply consolidator/modules/config/configurations.bal]: resolved

File: consolidator/consolidator_service.bal (hunk 7 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply consolidator/consolidator_service.bal:36]: resolved

File: consolidator/modules/kafkahealthcheck/kafkahealthcheck.bal (hunk 1 above)
Comment (vishwa-vyom) [reply consolidator/modules/kafkahealthcheck/kafkahealthcheck.bal]: resolved

File: hub/hub_service.bal (hunk 2 above)
Comment (vishwa-vyom) [reply hub/hub_service.bal:47]: resolved

File: hub/modules/config/configurations.bal (hunk 3 above)
Comment (vishwa-vyom) [reply hub/modules/config/configurations.bal:50]: resolved

File: hub/modules/config/configurations.bal (hunk 4 above)
Comment (vishwa-vyom) [reply hub/modules/config/configurations.bal]: resolved

File: hub/modules/inittopic/inittopic.bal (hunk 5 above)
Comment (vishwa-vyom) [reply hub/modules/inittopic/inittopic.bal]: As discussed lets change kafka health check also to be underscore separated.

**Review by vishwa-vyom — APPROVED** [review]

File: hub/modules/inittopic/inittopic.bal (hunk 5 above)
Comment (vishwa-vyom) [reply hub/modules/inittopic/inittopic.bal]: resolved

**Review by vadherhemant — APPROVED** [review]
needed for the upgrade to have these changes

**Review by Rakshitha650 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**